packages/graphql-language-service-server/src/GraphQLSchemaLoaderExtension.js

```diff
@@ -0,0 +1,26 @@
+import {GraphQLSchema} from 'graphql';
+
+export type GraphQLSchemaLoaderPackage = {
+  init: (loaderArgs: any) => GraphQLSchemaLoader,
+}
+
+export type GraphQLSchemaLoader = {
+  getSchema: () => Promise<?GraphQLSchema>,
+}
+
+export type GraphQLSchemaConfigData = {
+  loaderPackage: string,
+  loaderArgs: any,
+}
+
+export class GraphQLSchemaLoaderExtension {
+  _raw: GraphQLSchemaConfigData;
+
+  constructor(schemaConfig: GraphQLSchemaConfigData) {
+      this.raw = schemaConfig;
+  }
+
+  getSchemaLoader(): GraphQLSchemaLoader {
+    return require(this.raw.loaderPackage).init(this.raw.loaderArgs);
```

Comment:
Everything about this looks good to me. Love the idea as well. The only part about this that I'm not sure of is this line which you guys were already talking about. I'm not familiar with all the contexts the language server can be used in (for example a browser context might not have the `require` method), so when this is called at runtime it would need to work well for all contexts it's used in.

Having the package add itself to a global object is one way to go. Probably there are other approaches for this.

_Ps: If the language server always runs in a node context, then I would prefer the `require` to the global object approach._

Reply:
I think we should try and stay away from the `global` object and dynamic `require`s. All these concerns should be pushed up the stack so the calling application deals with them directly (and can then choose to do dynamic require vs not depending on what bundling they're using).

With that in mind, should `getSchemaLoader` be async?

Reply:
i think we are working towards allowing this to load behavior based on the platform - so in VSCode that would be via an extension, with node it should
 be an async file load not a require, and in the browser we will need to load it differently, probably dynamic import. What it needs is to dynamically decide which platform it's using and to attempt to load the schema loading plugin with the matching mechanism. @Neitsch and I were discussing this

Reply:
we have a fix for this now! :D

Reply:
@imolorhe as it turns out, webpack should handle the require method just fine, we will shift towards async imports as well, as should electron/etc, as import() is supported in node 9.7+. The other thing is this is on the server side, so this code won't be loaded by webpack.

@Neitsch if you want to update this for the new file resolution utilities we merged a week or two ago (they are in gls-utils now) they should ensure that this dynamic `require()` line works across environments.

also, is it possible we can use graphql-config to load the schema? looks like it has its own capability for this already. check out the latest 3.0.0 alpha, we will be adopting this soon!